### PR TITLE
Feat : Switched to a new working Implementation for APK/AAB signing

### PR DESCRIPTION
**Pull request by Yamenher**
I added my own Implementation from my X-Signer app which relies on ported libraries to android in java based on `ApkSigner` library and also used ported library to add support for JKS keys which is not supported by default by android, For detailed infos, view commits

### CODE REVIEW

File: app/src/main/java/mod/jbk/export/GetKeyStoreCredentialsDialog.java [hunk 1]
```diff
@@ -172,8 +172,8 @@ public boolean isForSigningWithTestkey() {
         /**
          * @return {@link #keyStorePassword}
          */
-        public String getKeyStorePassword() {
-            return keyStorePassword;
+        public static String getKeyStorePassword() {
+            return binding.etKeyPassword.getText().toString();
         }
```
Comment (aikrq) [inline app/src/main/java/mod/jbk/export/GetKeyStoreCredentialsDialog.java:177]: It's bad to get string from non static field for static function.

**Review by aikrq — CHANGES_REQUESTED** [review]

File: app/src/main/java/mod/jbk/export/GetKeyStoreCredentialsDialog.java (hunk 1 above)
Comment (Yamenher) [reply app/src/main/java/mod/jbk/export/GetKeyStoreCredentialsDialog.java:177]: didn't notice that sorry

**Comment by ilyassesalama** [comment]
Your changes show this error. Can you double-check if everything is alright? Also don't forget to pull my changes.

Screenshot 2024-11-07 at 03 27 16

**Comment by Yamenher** [comment]
> Your changes show this error. Can you double-check if everything is alright? Also don't forget to pull my changes.
> 
> Screenshot 2024-11-07 at 03 27 16

Ok I'll analyse the code

**Comment by Yamenher** [comment]
> Your changes show this error. Can you double-check if everything is alright? Also don't forget to pull my changes.
> 
> <img alt="Screenshot 2024-11-07 at 03 27 16" width="351" src="https://private-user-images.githubusercontent.com/46769766/383790396-fdf14105-bca1-4738-b4a6-6474abb1ea09.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5NjA0ODUsIm5iZiI6MTczMDk2MDE4NSwicGF0aCI6Ii80Njc2OTc2Ni8zODM3OTAzOTYtZmRmMTQxMDUtYmNhMS00NzM4LWI0YTYtNjQ3NGFiYjFlYTA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA3VDA2MTYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyNTAyYzFlMjIxNjVlMjNhODE3YzA5Y2E0ZjliYThiN2RlOTZhZTNhYTdhMjA4MzMyMDE1M2VjYmE0MjFmNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.E451uUyt0MiSjNKTlUXoeXlcU3VbitG4IxE8xojXaBA"
Looks like the problem is with your commits, cuz k never got this until i pulled your changes

**Comment by elfilibusterismo** [comment]
> Your changes show this error. Can you double-check if everything is alright? Also don't forget to pull my changes.
> 
> Screenshot 2024-11-07 at 03 27 16

Idk but, it doesn't seem like it caused by this pr

**Comment by Yamenher** [comment]
> > Your changes show this error. Can you double-check if everything is alright? Also don't forget to pull my changes.
> > Screenshot 2024-11-07 at 03 27 16
> 
> Idk but, it doesn't seem like it caused by this pr

@ilyassesalama @elfilibusterismo guys once i reverted to my unchanged commit, everything got back to work so if the issue from Ilyasse's PR
Screenshot_20241107-072219_Sketchware Pro

**Comment by ilyassesalama** [comment]
The issue is not from commits. It seems like it’s happening because I synced this PR code with the main branch.

**Comment by Yamenher** [comment]
> The issue is not from commits. It seems like it’s happening because I synced this PR code with the main branch. Well, if it’s not working after merging main with your PR, it won’t work vice-versa.

well did you see my screenshot? i didn't get any error upon running signer, idk why, but i have a feeling it's either from your changing of android.util.Base64 to java.util.Base64 or removing something i did for a reason and not just randomly

**Comment by Yamenher** [comment]
you can try the code without your commits to verify your self if what i said isn't that convincing

**Comment by ilyassesalama** [comment]
Can you try to idendity the issue if possible?

**Comment by Yamenher** [comment]
> Can you try to idendity the issue if possible?

I don't think that's possible, for some reason, logging isn't working at all :/
Weird

**Comment by ilyassesalama** [comment]
> > Thank you for finally fixing this very old issue,
> 
> No worries I did nothing 😉

Well, apparently it seems like it’s breaking some projects. Users can’t install build APKs. Any idea why this issue occurs? If you’re unable to solve the issue, I’ll revert this pull request.

**Comment by laymintu** [comment]
> Well, apparently it seems like it’s breaking some projects. Users can’t install built APKs when AppCompat is enabled. Any idea why this issue occurs? If you’re unable to solve the issue, I’ll revert this pull request.



This is why i can't install my 2 year project
🙂🙂🙂

**Comment by ilyassesalama** [comment]
> This is why i can't install my 2 year project 🙂🙂🙂

Can you try this? and let me know if it works.

https://github.com/user-attachments/assets/d3e71a3e-1bc7-4a92-adaf-3e2b1510ce14

**Comment by Yamenher** [comment]
> > Well, apparently it seems like it’s breaking some projects. Users can’t install built APKs when AppCompat is enabled. Any idea why this issue occurs? If you’re unable to solve the issue, I’ll revert this pull request.
> 
> This is why i can't install my 2 year project 🙂🙂🙂
> 
> 

umm, I think this is an old data issue, refresh project options and configuration and try again, I fully tested the signer before adding it fyi

**Comment by ilyassesalama** [comment]
This PR has been reverted. If you still want to include this feature to Sketchware Pro, solve the installation issue.

**Comment by ilyassesalama** [comment]
> What issue tho?

I already mentioned the issue, you can also check our Telegram group for more information.

**Comment by laymintu** [comment]
> > This is why i can't install my 2 year project 🙂🙂🙂
> 
> Can you try this? and let me know if it works.
> 
> https://github.com/user-attachments/assets/d3e71a3e-1bc7-4a92-adaf-3e2b1510ce14
> 
> 


Thanks for the video but it does not work i am waiting for the new version

**Pull request merged**